src/map.cpp

```diff
@@ -3508,15 +3508,10 @@ void map::bash_ter_furn( const tripoint &p, bash_params &params )
         spawn_items( p, item_group::items_from( bash->drop_group, calendar::turn ) );
     }
 
-    if( smash_ter && ter( p ) == t_open_air ) {
-        if( !zlevels ) {
-            // We destroyed something, so we aren't just "plugging" air with dirt here
-            ter_set( p, t_dirt );
-        } else {
-            tripoint below( p.xy(), p.z - 1 );
-            const auto roof = get_roof( below, params.bash_floor && ter( below ).obj().movecost != 0 );
-            ter_set( p, roof );
-        }
+    if( smash_ter && ter( p ) == t_open_air && zlevels ) {
```

Comment:
zlevels are always on, so you could remove this check.

Reply:
It's actually not. I get a crash after removing the `zlevels` check:

```
map::get_roof(tripoint const&, bool)
    …/src/map.cpp:3218
    map::bash_ter_furn(tripoint const&, bash_params&)
    …/src/map.cpp:3513 (discriminator 10)
    map::bash(tripoint const&, int, bool, bool, bool, vehicle const*)
    …/src/map.cpp:3565
    map::destroy(tripoint const&, bool)
    …/src/map.cpp:3641 (discriminator 1)
    MapExtras::mx_crater(map&, tripoint const&)
    …/src/map_extras.cpp:1530
    MapExtras::apply_function(string_id<map_extra> const&, map&, coords::coord_point<tripoint, (coords::origin)1, (coords::scale)1> const&)
    …/src/map_extras.cpp:3017
    MapExtras::apply_function(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, map&, coords::coord_point<tripoint, (coords::origin)1, (coords::scale)1> const&)
    …/src/map_extras.cpp:3074
    map::generate(tripoint const&, time_point const&)
    …/src/mapgen.cpp:239
    map::loadn(tripoint const&, bool, bool)
    …/src/map.cpp:6939
    map::loadn(point const&, bool, bool)
    …/src/map.h:1731
    map::load(coords::coord_point<tripoint, (coords::origin)1, (coords::scale)1> const&, bool, bool)
    …/src/map.cpp:6598
    game::load_map(coords::coord_point<tripoint, (coords::origin)1, (coords::scale)1> const&, bool)
    …/src/game.cpp:735
    game::place_player_overmap(coords::coord_point<tripoint, (coords::origin)1, (coords::scale)2> const&)
    …/src/game.cpp:9667
```